### PR TITLE
Fwmt 430 - Part 1

### PR DESCRIPTION
**Pull request by spdiaz1000**
Ive refactored some code and I want to get it into master as soon as possible as its beginning to get complicated to keep up to date with master

### CODE REVIEW

File: src/main/java/uk/gov/ons/fwmt/job_service/rest/client/impl/JobResourceServiceClientImpl.java [hunk 1]
```diff
@@ -1,75 +1,65 @@
...
-      log.debug("existsByTmJobId: JobDto found");
-    } else {
-      log.debug("Not found");
-    }
+    final Optional<JobDto> jobDto = findByTmJobId(tmJobId);
     return jobDto.isPresent();
   }
 
   @Override
   public boolean existsByTmJobIdAndLastAuthNo(String tmJobId, String lastAuthNo) {
```
Comment (soundargit) [inline src/main/java/uk/gov/ons/fwmt/job_service/rest/client/impl/JobResourceServiceClientImpl.java]: Functionality  is missing  to search based on  jobid and last authno.

File: src/main/java/uk/gov/ons/fwmt/job_service/service/impl/CSVParsingServiceImpl.java [hunk 2]
```diff
@@ -120,18 +120,16 @@ protected String constructTmJobId(CSVRecord record, LegacySampleSurveyType surve
...
       return fieldPeriod.getEndDate();
     } else {
-      throw FWMTCommonException.makeUnknownFieldPeriodException(fp);
+      throw FWMTCommonException.makeUnknownFieldPeriodException(stage);
     }
   }
 
-  // TODO replace with a function that calculates the GFF date deterministically
-  // Currently, it falls back on checking the database
   public LocalDate convertToGFFDate(String stage) {
```
Comment (soundargit) [inline src/main/java/uk/gov/ons/fwmt/job_service/service/impl/CSVParsingServiceImpl.java]: we can merge the call. do we need two methods?

File: src/main/java/uk/gov/ons/fwmt/job_service/service/impl/JobProcessor.java [hunk 3]
```diff
@@ -1,95 +1,100 @@
...
-  ) {
-    this.fileIngestService = fileIngestService;
-    this.csvParsingService = csvParsingService;
-    this.userResourceService = userResourceService;
-    this.tmJobConverterService = tmJobConverterService;
-    this.jobResourceService = jobResourceService;
-    this.tmService = tmService;
+  private TMService tmService;
+  
+  public JobProcessor(CSVParsingService csvParsingService,
```
Comment (soundargit) [inline src/main/java/uk/gov/ons/fwmt/job_service/service/impl/JobProcessor.java]: why do we need a constructor if everything is auto wired from instance?

File: src/main/java/uk/gov/ons/fwmt/job_service/service/impl/FileIngestServiceImpl.java [hunk 4]
```diff
@@ -25,115 +31,32 @@
...
-        tla = LegacySampleSurveyType.GFF;
-        break;
-      default:
-        throw new IllegalArgumentException("File had an unrecognized TLA of " + tlaString);
+  
+  public SampleSummaryDTO validateSampleFile(File file) throws IOException{
+    SampleFilenameComponents filename = SampleFileUtils.buildSampleFilenameComponents(file);
+    final Reader reader = new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8);
+    
+    Iterator<CSVParseResult<LegacySampleIngest>> csvRowIterator = csvParsingService.parseLegacySample(reader, filename.getTla());
```
Comment (soundargit) [inline src/main/java/uk/gov/ons/fwmt/job_service/service/impl/FileIngestServiceImpl.java]: Above 3 lines are called twice, do we need this? possible to make this call once saving some time.

Comment (spdiaz1000) [reply src/main/java/uk/gov/ons/fwmt/job_service/service/impl/FileIngestServiceImpl.java]: As discussed Ill leave this as is for the moment until we relook at this area

**Pull request merged**